src/homepage/router.js

```diff
@@ -7,15 +7,15 @@ import docs from './docs';
 import enToSnake from './utils/enToSnake';
 import Home from './pages/Home';
 
-let componentsChildren = [];
-docs.map(components => {
-  if (components.children && components.children.length) {
-    components.children.forEach(c => componentsChildren.push({
-      path: enToSnake(c.en),
-      component: c.component
-    }));
-  }
-});
+let componentChildren = docs.reduce((componentChildrenSoFar, component) =>
+  component.children
+    ? componentChildrenSoFar.concat(component.children.map(child => {
+      return {
+        path: enToSnake(child.en),
+        component: child.component
+      };
+    }))
+    : componentChildrenSoFar, []);
```

Comment:
1. Indeed the readability of `reduce` function is not good but if you understand it, it will be simple.
2. At the first time, I have the same viewpoint but difference after I find out these two testings https://jsperf.com/concatperftest/6 & https://jsperf.com/javascript-array-concat-vs-push